REFERENCE.md

```diff
@@ -0,0 +1,240 @@
+# Learnosity Ruby-SDK: Reference Guide
+
+## Usage
+
+Some usage examples can be found in the `examples/` subdirectory.
+
+### Init
+
+This class generates and signs init options for all supported APIs. Its
+constructor takes four mandatory arguments, and one optional argument.
+
+- `service`: the name of the API to sign initialisation options for,
+- `security_packet`: a hash with at least the `consumer_key`, optionally the
+  `domain` and `timestamp`; `user_id` is also mandatory for Questions API,
+- `consumer_secret`
+- `request`: the request you want to get a signature for
+- `action` [Data API only]: the action of your request, either `get` or `post`
+
+The `Init#generate` method can then be used to generate the options. By default,
+it will generate a JSON string. It however takes one parameter, `encode` which,
+if false, will simply return a native Ruby Hash with the signed options.
+
+Given your consumer key and secret, and the request you want to run against,
+say, the `items` API, you just need to instantiate the
+`Learnosity::Sdk::Request::Init`, and call its `generate` method.
+
+```ruby
+require "learnosity/sdk/request/init"
+
+security_packet = {
+    # XXX: This is a Learnosity Demos consumer; replace it with your own consumer key. Set values in application.rb.
+    'consumer_key'   => Rails.configuration.consumer_key,
+	'domain'         => 'localhost'
```

Comment:
The indent is weird here. There might be a mix of tabs and spaces.

Reply:
Fixed. I changed a tab to spaces.